flow/e2e/test_utils.go

```diff
@@ -129,58 +128,78 @@ func EnvEqualTables(env *testsuite.TestWorkflowEnvironment, suite e2eshared.RowS
 	EnvEqualRecordBatches(t, env, pgRows, rows)
 }
 
-func SetupCDCFlowStatusQuery(env *testsuite.TestWorkflowEnvironment,
-	connectionGen FlowConnectionGenerationConfig,
+func EnvWaitForEqualTables(
+	env *testsuite.TestWorkflowEnvironment,
+	suite e2eshared.RowSource,
+	reason string,
+	table string,
+	cols string,
 ) {
-	// wait for PeerFlowStatusQuery to finish setup
-	// sleep for 5 second to allow the workflow to start
-	time.Sleep(5 * time.Second)
-	for {
-		response, err := env.QueryWorkflow(
-			shared.CDCFlowStateQuery,
-			connectionGen.FlowJobName,
-		)
-		if err == nil {
-			var state peerflow.CDCFlowWorkflowState
-			err = response.Get(&state)
-			if err != nil {
-				slog.Error(err.Error())
-			} else if state.CurrentFlowState == protos.FlowStatus_STATUS_RUNNING {
-				break
-			}
-		} else {
-			// log the error for informational purposes
-			slog.Error(err.Error())
+	suite.T().Helper()
+	EnvWaitForEqualTablesWithNames(env, suite, reason, table, table, cols)
+}
+
+func EnvWaitForEqualTablesWithNames(
+	env *testsuite.TestWorkflowEnvironment,
+	suite e2eshared.RowSource,
+	reason string,
+	srcTable string,
+	dstTable string,
+	cols string,
+) {
+	t := suite.T()
+	t.Helper()
+
+	EnvWaitFor(t, env, 3*time.Minute, reason, func() bool {
+		t.Helper()
+
+		suffix := suite.Suffix()
+		pool := suite.Pool()
+		pgRows, err := GetPgRows(pool, suffix, srcTable, cols)
+		if err != nil {
+			return false
 		}
-		time.Sleep(1 * time.Second)
-	}
+
+		rows, err := suite.GetRows(dstTable, cols)
+		if err != nil {
+			return false
+		}
+
+		return e2eshared.CheckEqualRecordBatches(t, pgRows, rows)
+	})
 }
 
-func NormalizeFlowCountQuery(env *testsuite.TestWorkflowEnvironment,
+func SetupCDCFlowStatusQuery(t *testing.T, env *testsuite.TestWorkflowEnvironment,
 	connectionGen FlowConnectionGenerationConfig,
-	minCount int,
 ) {
-	// wait for PeerFlowStatusQuery to finish setup
-	// sleep for 5 second to allow the workflow to start
-	time.Sleep(5 * time.Second)
+	t.Helper()
+	// errors expected while PeerFlowStatusQuery is setup
+	counter := 0
 	for {
+		time.Sleep(time.Second)
+		counter++
 		response, err := env.QueryWorkflow(
 			shared.CDCFlowStateQuery,
 			connectionGen.FlowJobName,
 		)
 		if err == nil {
 			var state peerflow.CDCFlowWorkflowState
 			err = response.Get(&state)
-			if err != nil {
+			if err == nil {
+				if state.CurrentFlowState == protos.FlowStatus_STATUS_RUNNING {
+					return
+				}
+			} else {
 				slog.Error(err.Error())
-			} else if len(state.NormalizeFlowStatuses) >= minCount {
-				break
 			}
-		} else {
+		} else if counter > 15 {
```

Comment:
Is it worth having a parameter called `setupTimeout` or something instead of having a hard value of 15 seconds?

Reply:
No, because setup should always take the same amount of time